### PR TITLE
OSDOCS-5022:Added missing step for Selecting an AMI section

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.10+

Issue:
[OSDOCS-5022](https://issues.redhat.com/browse/OSDOCS-5022)

Link to docs preview:
[Selecting an Azure Marketplace image](https://56119--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-user-infra.html#installation-azure-marketplace-subscribe_installing-azure-user-infra)

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: installing/installing_azure/installing-azure-customizations.adoc [hunk 1]
```diff
@@ -26,6 +26,11 @@ include::modules/ssh-agent-using.adoc[leveloffset=+1]
 
 include::modules/installation-azure-marketplace-subscribe.adoc[leveloffset=+1]
 
+[role="_additional-resources"]
```
Comment (mjpytlak) [inline installing/installing_azure/installing-azure-customizations.adoc:29]: @dfitzmau This xref is out of context. This is an IPI assembly. You are cross referencing content for a UPI install. Please remove.

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 2]
```diff
@@ -129,7 +129,47 @@ ifdef::ipi[]
...
+      "sku": "rh-ocp-worker",
+      "version": "4.8.2021122100"
+  }
+}
+...
+----
+
+. Add a plan to the ARM template for your worker nodes.
++
+[IMPORTANT]
```
Comment (mjpytlak) [inline modules/installation-azure-marketplace-subscribe.adoc:152]: While true, I suggest removing this admonition. You are explicitly telling users to complete a required step. No need to tell them what happens if they do not follow the doc. ;)

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 3]
```diff
@@ -129,7 +129,47 @@ ifdef::ipi[]
...
+      "offer": "rh-ocp-worker",
+      "publisher": "redhat",
+      "sku": "rh-ocp-worker",
+      "version": "4.8.2021122100"
+  }
+}
+...
+----
+
+. Add a plan to the ARM template for your worker nodes.
```
Comment (mjpytlak) [inline modules/installation-azure-marketplace-subscribe.adoc:150]: This is also related to using the Azure Resource Manager ARM templates to deploy worker nodes. I suggest something along the lines of

5. Record the image details of your offer. If you use the Azure Resource Manager (ARM) template to deploy your worker nodes:
     a. Update `storageprofile.imageReferences`....
     b. Specify a `plan` for the virtual machines....

**Review by mjpytlak — COMMENTED** [review]
@dfitzmau Bit of feedback on this.

File: installing/installing_azure/installing-azure-customizations.adoc (hunk 1 above)
Comment (dfitzmau) [reply installing/installing_azure/installing-azure-customizations.adoc:29]: OK. 

I'm making changes to a module (`installation-azure-marketplace-subscribe`) in the assembly `installing-azure-user-infra.adoc . I have seen many references to UPI in this assembly. Just want to double-check with you that this assembly covers IPI only? 

My intention was to steer the user to a complete file structure without having to possibly duplicate the effort.

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 2 above)
Comment (dfitzmau) [reply modules/installation-azure-marketplace-subscribe.adoc]: Yeah, good point. Removed. 

I misunderstood that the added step was somewhat assumed based on how it was missing from the original steps.

File: installing/installing_azure/installing-azure-customizations.adoc (hunk 1 above)
Comment (mjpytlak) [reply installing/installing_azure/installing-azure-customizations.adoc:29]: Yes. This file is an IPI installation topic. The xref is not required.

**Comment by dfitzmau** [comment]
Hello @newera-begins 

Would you be OK to review this PR? Preview link available in the PR description box.

**Comment by dfitzmau** [comment]
PR blocked. See comment on Jira.

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 4]
```diff
@@ -129,7 +129,36 @@ ifdef::ipi[]
...
+      "publisher": "redhat",
+      "sku": "rh-ocp-worker",
+      "version": "4.8.2021122100"
+  }
+}
+...
+  "plan" : {
+    "name": "rh-ocp-worker",
+    "product": "rh-ocp-worker",
+    "publisher": "redhat-limited"
```
Comment (jinyunma) [inline modules/installation-azure-marketplace-subscribe.adoc:154]: `redhat-limited` -> `redhat`, to keep consistent with storageProfile

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 5]
```diff
@@ -129,7 +129,36 @@ ifdef::ipi[]
...
+  },
+  "dependsOn" : [
+    "[concat('Microsoft.Network/networkInterfaces/', concat(variables('vmNames')[copyIndex()], '-nic'))]"
+  ],
+...
```
Comment (jinyunma) [inline modules/installation-azure-marketplace-subscribe.adoc:159]: Checked 06_worker.json file, better to move "plan"..."dependsOn"  in front of storageProfile to easily find the place to update.

Comment (dfitzmau) [reply modules/installation-azure-marketplace-subscribe.adoc:159]: Thanks. I moved this section. Does the `],` syntax still stand on line 149 (on updated file)?

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 4 above)
Comment (dfitzmau) [reply modules/installation-azure-marketplace-subscribe.adoc]: Updated.

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 5 above)
Comment (jinyunma) [reply modules/installation-azure-marketplace-subscribe.adoc]: no, I checked 06_worker.json, it should be correct.

seems 'storageProfille' is not at the same level with "plan" and "dependsOn", suggest to check the content of 06_worker.json in section "Creating additional worker machines in Azure" and adjust the format here.

**Comment by jinyunma** [comment]
LGTM

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 6]
```diff
@@ -129,7 +129,39 @@ ifdef::ipi[]
 . Record the image details of your offer. You must update the `compute` section in the `install-config.yaml` file with values for `publisher`, `offer`, `sku`, and `version` before deploying the cluster.
 endif::ipi[]
 ifdef::upi[]
-. Record the image details of your offer. If you use the Azure Resource Manager (ARM) template to deploy your worker nodes, you can update `storageProfile.imageReference` by deleting the `id` parameter and adding the `offer`, `publisher`, `sku`, and `version` parameters using the values from your offer.
+. Record the image details of your offer. If you use the Azure Resource Manager (ARM) template to deploy your worker nodes:
```
Comment (eohartman) [inline modules/installation-azure-marketplace-subscribe.adoc:132]: What are the steps if you don't use the Azure Resource Manager template to deploy your worker nodes?

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 7]
```diff
@@ -129,7 +129,39 @@ ifdef::ipi[]
...
++
+.. Update `storageProfile.imageReference` by deleting the `id` parameter and adding the `offer`, `publisher`, `sku`, and `version` parameters by using the values from your offer.
+.. Specify a `plan` for the virtual machines (VMs). 
++
+.Sample `06_workers.json` ARM template with an updated `storageProfile.imageReference` object and a specified `plan`
```
Comment (eohartman) [inline modules/installation-azure-marketplace-subscribe.adoc:137]: Change "Sample" to "Example." And this sentence needs a colon at the end of it.

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 8]
```diff
@@ -129,7 +129,39 @@ ifdef::ipi[]
...
+----
+...
+  "plan" : {
+    "name": "rh-ocp-worker",
+    "product": "rh-ocp-worker",
+    "publisher": "redhat"
+  },
+  "dependsOn" : [
+    "[concat('Microsoft.Network/networkInterfaces/', concat(variables('vmNames')[copyIndex()], '-nic'))]"
+  ],
```
Comment (eohartman) [inline modules/installation-azure-marketplace-subscribe.adoc:149]: On the code block in the preview, some of the characters (namely, "..." and "," are showing up in red. Is this intentional? (I'm still learning, so maybe it is!?)

**Review by eohartman — CHANGES_REQUESTED** [review]

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 6 above)
Comment (dfitzmau) [reply modules/installation-azure-marketplace-subscribe.adoc:132]: The doc update request targets a specific update. Covering other routes is out of scope.

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 7 above)
Comment (dfitzmau) [reply modules/installation-azure-marketplace-subscribe.adoc:137]: Updated to "Example". A colon is not required for code block titles. See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 8 above)
Comment (dfitzmau) [reply modules/installation-azure-marketplace-subscribe.adoc:149]: Well spotted, @eohartman . I removed the default substitutions, so these highlighted red entities no longer exist.

**Pull request merged**